src/4.3_to_5.0/accumulateMigratedKpiTitles.ts

```
@@ -107,35 +113,44 @@ export function getMigratedKpiTitles(
-    const memberUniqueNames: string[] = [];
+    try {
+      const memberUniqueNames: string[] = [];
 
-    ordinalFields.forEach((field) => {
-      if (field.type === "hierarchy") {
-        const hierarchyUniqueName = quote(
-          field.dimensionName,
-          field.hierarchyName,
-        );
-        const namePath = tuple[hierarchyUniqueName];
-        if (namePath) {
-          memberUniqueNames.push(
-            `${hierarchyUniqueName}.${quote(...namePath)}`,
+      ordinalFields.forEach((field) => {
+        if (field.type === "hierarchy") {
+          const hierarchyUniqueName = quote(
+            field.dimensionName,
+            field.hierarchyName,
           );
+          const namePath = tuple[hierarchyUniqueName];
+          if (namePath) {
+            memberUniqueNames.push(
+              `${hierarchyUniqueName}.${quote(...namePath)}`,
+            );
+          }
         }
+      });
+
+      if (measuresPositionInTuple > -1) {
+        const measureName = tuple[quote("Measures", "Measures")][0];
+        memberUniqueNames.splice(
+          measuresPositionInTuple,
+          0,
+          quote("Measures", measureName),
+        );
       }
-    });
-
-    if (measuresPositionInTuple > -1) {
-      const measureName = tuple[quote("Measures", "Measures")][0];
-      memberUniqueNames.splice(
-        measuresPositionInTuple,
-        0,
-        quote("Measures", measureName),
-      );
-    }
 
-    const tupleKey = memberUniqueNames.join(",");
-    migratedTitles[tupleKey] = title;
+      const tupleKey = memberUniqueNames.join(",");
+      migratedTitles[tupleKey] = title;
+    } catch (error) {
+      errors.push(error as Error);
+    }
   });
 
-  return migratedTitles;
+  if (errors.length > 0) {
+    throw new Error(
+      `One or more errors occurred while migrating the titles for the kpi named ${legacyKpiState.name}: ` +
+        errors.map((e) => e.message).join(", "),
+    );
+  }
```

This function "swallows" any error until all `legacyTitles` have been attempted to be migrated. This throws at the end to ensure that the kpi widget that is being migrated is marked as `partially migrated`.